File: 3950-maximum-k-to-sort-a-permutation/3950-maximum-k-to-sort-a-permutation.py
class Solution:
    def sortPermutation(self, nums: List[int]) -> int:
        n = len(nums)
        ans = -1
        for i in range(n):
            if nums[i] != i:
                ans = i
                break
        if ans == -1:
            return 0
        for i in range(n):
            if nums[i] != i:
                ans = ans & i
        
        return ans

File: 3950-maximum-k-to-sort-a-permutation/3950-maximum-k-to-sort-a-permutation.cpp
class Solution {
public:
    int sortPermutation(vector<int>& nums) {
        int n = nums.size();
        int ans = -1;
        for(int i = 0; i < n; i++)
        {
            if(nums[i] != i)
            {
                ans = i;
                break;
            }
        }

        if(ans == -1) return 0;
        for(int i = 0; i < n; i++)
        {
            if(nums[i] != i)
            {
                ans = ans & i;
            }
        }
        return ans;
    }
};